File: GeneratedEntityReaderClasses/Components/HotspotComponent.cs
namespace GeneratedNoitaClasses;
public class HotspotComponent : NoitaComponentBase,  iNoitaType<HotspotComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaCVector2<NoitaFloat> offset { get; set; } = new();
	public NoitaBool transform_with_scale { get; set; } = new();
	public NoitaString sprite_hotspot_name { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		offset.Read(s);
		transform_with_scale.Read(s);
		sprite_hotspot_name.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		offset.Write(s);
		transform_with_scale.Write(s);
		sprite_hotspot_name.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/HotspotComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaCVector2
from BasicNoitaTypes import NoitaFloat

from NoitaStream import NoitaStream
@dataclass
class HotspotComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    offset:NoitaCVector2 = field(default_factory=lambda: NoitaCVector2(NoitaFloat))
    transform_with_scale:NoitaBool = field(default_factory=NoitaBool)
    sprite_hotspot_name:NoitaString = field(default_factory=NoitaString)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.offset.Read(s)
        self.transform_with_scale.Read(s)
        self.sprite_hotspot_name.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.offset.Write(s)
        self.transform_with_scale.Write(s)
        self.sprite_hotspot_name.Write(s)

